solana/programs/token-bridge-relayer/src/processor/outbound.rs

```diff
@@ -0,0 +1,491 @@
+    let truncated_amount = token_bridge::truncate_amount(transferred_amount, mint.decimals);
+    require!(
+        truncated_amount > 0,
+        TokenBridgeRelayerError::ZeroBridgeAmount
+    );
+
+    // Normalize the dropoff amount:
+    //FIXME: it should not be the mint decimals
+    let normalized_dropoff_amount =
+        token_bridge::normalize_amount(gas_dropoff_amount, mint.decimals)
+            .try_into()
+            .map_err(|_| TokenBridgeRelayerError::Overflow)?;
+    require!(
+        gas_dropoff_amount == 0 || normalized_dropoff_amount > 0,
+        TokenBridgeRelayerError::InvalidToNativeAmount
+    );
+
+    Ok((truncated_amount, normalized_dropoff_amount))
+}
+
+fn token_bridge_transfer_native(
+    ctx: &mut Context<OutboundTransfer>,
+    transferred_amount: u64,
+    recipient_chain: u16,
+    relayer_message: &RelayerMessage,
+) -> Result<()> {
+    let config_seed = &[
+        TbrConfigState::SEED_PREFIX.as_ref(),
+        &[ctx.bumps.tbr_config],
+    ];
+
+    let token_bridge_custody = ctx
+        .accounts
+        .token_bridge_custody
+        .as_ref()
+        .expect("We have checked that before");
+    let token_bridge_custody_signer = ctx
+        .accounts
+        .token_bridge_custody_signer
+        .as_ref()
+        .expect("We have checked that before");
+
+    token_bridge::transfer_native_with_payload(
+        CpiContext::new_with_signer(
+            ctx.accounts.token_bridge_program.to_account_info(),
+            token_bridge::TransferNativeWithPayload {
+                payer: ctx.accounts.payer.to_account_info(),
+                config: ctx.accounts.token_bridge_config.to_account_info(),
+                from: ctx.accounts.temporary_account.to_account_info(),
+                mint: ctx.accounts.mint.to_account_info(),
+                custody: token_bridge_custody.to_account_info(),
+                authority_signer: ctx.accounts.token_bridge_authority_signer.to_account_info(),
+                custody_signer: token_bridge_custody_signer.to_account_info(),
+                wormhole_bridge: ctx.accounts.wormhole_bridge.to_account_info(),
+                wormhole_message: ctx.accounts.wormhole_message.to_account_info(),
+                wormhole_emitter: ctx.accounts.token_bridge_emitter.to_account_info(),
+                wormhole_sequence: ctx.accounts.token_bridge_sequence.to_account_info(),
+                wormhole_fee_collector: ctx.accounts.fee_recipient.to_account_info(),
+                clock: ctx.accounts.clock.to_account_info(),
+                sender: ctx.accounts.tbr_config.to_account_info(),
+                rent: ctx.accounts.rent.to_account_info(),
+                system_program: ctx.accounts.system_program.to_account_info(),
+                token_program: ctx.accounts.token_program.to_account_info(),
+                wormhole_program: ctx.accounts.wormhole_program.to_account_info(),
+            },
+            &[
+                config_seed,
```

Comment:
The config seeds here are wrong. Turns out that there were 3 different config accounts in the original TBR impl and I mistakenly thought the RedeemerConfig played a dual role.

In reality there are:
* [OwnerConfig](https://github.com/wormhole-foundation/example-token-bridge-relayer/blob/main/solana/programs/token_bridge_relayer/src/state/owner_config.rs) - seed: "owner"
* [RedeemerConfig](https://github.com/wormhole-foundation/example-token-bridge-relayer/blob/d9d17254dae48c985fe6b58e2987e2135d1e8c65/solana/programs/token_bridge_relayer/src/state/redeemer_config.rs#L20) - seed: redeemer (as demanded by the token bridge)
* and finally: [SenderConfig](https://github.com/wormhole-foundation/example-token-bridge-relayer/blob/main/solana/programs/token_bridge_relayer/src/state/sender_config.rs) - seed prefix: "sender"

In our case, ower and redeemer config are now merged and we are missing the equivalent of the sender config, which is required here. Given the current implementation, this causes an authority error because the PDA is derived from "redeemer" rather than "sender". So we need a new, separate PDA for signing outbound transfers with the token bridge, as was the case with the old impl.

Reply:
This is garbage. The internal implementation is leaking, and I need to create useless PDAs only to satisfy it.

Reply:
I added the seed